src/graph_legend.js

```diff
@@ -72,23 +72,21 @@ export function GraphLegend(targetElem, config, onVisibilityChange, onNewSelecti
                selectedGraphIndex    = $(this).attr('graph'),
                selectedFieldIndex    = $(this).attr('field');
 
-           if(!e.altKey) {
-               config.selectedFieldName     = config.getGraphs()[selectedGraphIndex].fields[selectedFieldIndex].friendlyName;
-               config.selectedGraphIndex    = selectedGraphIndex;
-               config.selectedFieldIndex    = selectedFieldIndex;
-               if (onNewSelectionChange) {
-                   onNewSelectionChange();
-               }
-           } else { // toggle the grid setting
-               var graphs = config.getGraphs();
-               for(var i=0; i<graphs[selectedGraphIndex].fields.length; i++) {
-                  graphs[selectedGraphIndex].fields[i].grid = ((i==selectedFieldIndex)?(!graphs[selectedGraphIndex].fields[i].grid):false);
-               };
-               if (onNewGraphConfig) {
-                   onNewGraphConfig(graphs);
-               }
-           };
-           e.preventDefault();
+            if(!e.altKey) {
+				const selectedFieldName = config.getGraphs()[selectedGraphIndex].fields[selectedFieldIndex].friendlyName;
+				if (config.selectedFieldName != selectedFieldName) {
+					config.selectedFieldName     = selectedFieldName;
+					config.selectedGraphIndex    = selectedGraphIndex;
+					config.selectedFieldIndex    = selectedFieldIndex;
+					if (onNewSelectionChange) {
+						onNewSelectionChange();
+					}
+				}
+				else {
+					onNewSelectionChange(true);
+				}
+            }
```

Comment:
- tabs need to be spaces :) 
- else clause should follow on same line as closing bracket

```suggestion
            if (!e.altKey) {
                const selectedFieldName = config.getGraphs()[selectedGraphIndex].fields[selectedFieldIndex].friendlyName;
                if (config.selectedFieldName != selectedFieldName) {
                    config.selectedFieldName     = selectedFieldName;
                    config.selectedGraphIndex    = selectedGraphIndex;
                    config.selectedFieldIndex    = selectedFieldIndex;
                   if (onNewSelectionChange) {
                       onNewSelectionChange();
                    }
                } else {
                   onNewSelectionChange(true);
                }
            }
```